doc/asset-contracts-cbor.md

```diff
@@ -0,0 +1,194 @@
+  * Do not display, or accept user entry of, values for other assets.
+  * Treat reissuance tokens specially, as always having precision 0 (that is, atomic indivisible units.)
+
+* Registry changes:
+  * Accept registraton of CBOR contracts.
+  * Provide for verification of CBOR contracts against the spec.
+  * Stop accepting new JSON contracts (perhaps with a "deprecation phase" where they are discouraged somehow).
+  * For convenience, allow retrieval of contracts either in CBOR format, or converted to JSON as described later in this document.
+
+* Tooling:
+  * `rust-elements`: support creating, inspecting, and verifying both old and new asset contracts
+  * `hal-elements`: support creating, inspecting, and verifying both old and new asset contracts
+  * TBD: Usable tooling for registering assets with the registry, and issuing assets on the blockchain.
+  * For users needing generic tooling for working with CBOR data: https://github.com/intel/tinycbor, http://cbor.me/
+
+## Contract format
+
+A version 1 asset contract consists of: a single-byte version field (which must be 1), followed by a CBOR-encoded array with 3 elements:
+* The precision, encoded as an unsigned integer from 0 to 8.
+* The ticker, encoded as a string, 3-5 characters from the character set "[a-zA-Z.-]" (that is, upper and lowercase letters, dot, and dash.)
+* A map containing other fields, all of which are OPTIONAL:
+  * (The field names `"precision"`, `"ticker"`, and `"entity"` are reserved for backwards compatibility, and may not be used.)
+  * `"name"`, a string containing ASCII characters, length 1 to 255 (inclusive)
+  * `"issuer_pubkey"`, a byte-string containing a secp256k1 public key in compressed form.
+  * `"domain"`, a domain name belonging to the issuer
+  * Any other fields desired by the issuer, subject to the restrictions below.
+    * For issuers who wish to make use of the onchain contract distribution feature in `OP_RETURN` outputs, attention should be paid to the size of the contract. For ancillary data which should be verifiable, but need not be distributed to all users in-band, it is recommended to store a hash of the data in a field of bytestring type, rather than the data itself.
+
+Further restrictions:
+* The entire contract, including the version byte and the entire CBOR structure, MUST be at most 256 bytes in length.
+  * This will be enforced as a standardness/policy rule, for contracts encoded in `OP_RETURN` outputs.
+  * Consumers SHOULD reject overlong contracts, but are not required to do so.
+* Some additional restrictions are imposed on the CBOR data, for security, ease of parsing, and (limited) JSON compatibility:
+  * The CBOR must be "well-formed" and "valid", and decoding must be done in "strict mode" if possible, as defined in RFC 7049:
+    * Maps MUST NOT have duplicate keys.
+    * Strings MUST be valid UTF-8.
```

Comment:
Domain names should be ASCII, punycode etc to mitigate https://en.wikipedia.org/wiki/IDN_homograph_attack